openapi.yaml

```diff
@@ -7807,6 +7807,667 @@ paths:
+      - url: https://api.linode.com/v4beta
+      tags:
+      - Networking
+      summary: Update Firewall Rules
+      description: |
+        Updates the inbound and outbound Rules for a Firewall. Using this endpoint will
+        replace all of a Firewall's ruleset with the Rules specified in your request.
+
+        **Beta**: This endpoint is in beta. Please make sure to prepend all requests with
+        `/v4beta` instead of `/v4`, and be aware that this endpoint may receive breaking
+        updates in the future.  This notice will be removed when this endpoint is out of
+        beta.
+      operationId: updateFirewallRules
+      x-linode-cli-action: rules-update
+      security:
+      - personalAccessToken: []
+      - oauth:
+        - firewall:read_write
+      requestBody:
+        description: The Firewall Rules information to update.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Firewall/properties/rules'
+      responses:
+        '200':
+          description: Firewall Rules updated successfully.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Firewall/properties/rules'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      x-code-samples:
+      - lang: Shell
+        source: >
+          curl -H "Content-Type: application/json" \
+              -H "Authorization: Bearer $TOKEN" \
+              -X PUT -d '{
+                  "inbound": [
+                    {
+                      "protocol": "TCP",
+                      "ports": "22, 80, 8080, 443"
+                      "addresses": {
+                        "ipv4": [
+                          "192.0.2.1",
+                          "192.0.2.0/24"
+                        ],
+                        "ipv6": [
+                          "2001:DB8::/32"
+                        ]
+                      }
+                    }
+                  ],
+                  "outbound": [
+                      "protocol": "TCP",
+                      "ports": "49152-65535"
+                      "addresses": {
+                        "ipv4": [
+                          "192.0.2.1",
+                          "192.0.2.0/24"
+                        ],
+                        "ipv6": [
+                          "2001:DB8::/32"
+                        ]
+                      }
+                    }
+                  ]
+                }' \
+              https://api.linode.com/v4beta/networking/firewalls/123/rules
+      - lang: CLI
+        source: >
+          linode-cli firewalls rules-update 123 \
+            --inbound '[{"protocol": "TCP", "ports": "22, 80, 8080, 443", "addresses": {"ipv4": ["192.0.2.1", "192.0.2.0/24"], "ipv6": ["2001:DB8::/32"]}}]' \
+            --outbound '[{"protocol": "TCP", "ports": "49152-65535", "addresses": {"ipv4": ["192.0.2.1", "192.0.2.0/24"], "ipv6": ["2001:DB8::/32"]}}]'
```

Comment:
@leslitagordita just leaving this here too; these are the lines that might need some work.

@Dorthu I'm not quite sure how to represent dictionary arguments like this in the CLI